modules/4337/contracts/Safe4337Module.sol

```diff
@@ -212,9 +212,21 @@ contract Safe4337Module is IAccount, HandlerContext, CompatibilityFallbackHandle
     }
 
     /**
-     * @dev Checks if the signatures length is correct and does not contain additional bytes. The function does not
+     * @notice Checks if the signatures length is correct and does not contain additional bytes. The function does not
      * check the integrity of the signature encoding, as this is expected to be checked by the {Safe} implementation
      * of {checkSignatures}.
+     * @dev Safe account has two types of signatures: EOA and Smart Contract signatures. While the EOA signature is
+     * fixed in size, the Smart Contract signature can be of arbitrary length. Safe encodes the Smart Contract
+     * signature length in the signature data. If appropriate length checks are not performed during the signature
+     * verification then a malicious bundler can pad additional bytes to the signatures data and make the account pay
+     * more gas than needed for user operation validation and reach the verificationGasLimit.
+     * `_checkSignaturesLength` function checks for the presence of any padded bytes to the `signature` data.
+     * However, there is an edge case that `_checkSignaturesLength` function cannot detect.
```

Comment:
Here is where I think the leading sentence is most relevant - the kinds of signatures and specifically the smart contract signature matter for this edge case.

Reply:
https://github.com/safe-global/safe-modules/pull/471/commits/3dff65b5656955f1b83d40b27641f72c5d864e97